### PR TITLE
[MIG] contract: Migration to 14.0

### PR DESCRIPTION
**Pull request by ozono**
contract migration

### CODE REVIEW

**Review by hparfr — APPROVED** [review]
LGTM

**Comment by pedrobaeza** [comment]
We have added a lot of changes in 12.0/13.0 (some of them still ongoing), so I think this should be redone with them. I understand that may be to ask too much to include them and rework the migration, so we can take that work if you can't afford it.

**Comment by ozono** [comment]
I'll do it. No prob.

**Comment by pedrobaeza** [comment]
OK, thanks. I'll tell you when we finish to port all the changes to v13. They are basically:

- Portal feature
- Modification tracking

**Comment by newtratip** [comment]
@ozono 13.0 has many changed, could you include them ?

**Comment by ozono** [comment]
> @ozono 13.0 has many changed, could you include them ?

I'm waiting for the signal from @pedrobaeza

File: contract/data/contract_cron.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding='UTF-8' ?>
```
Comment (newtratip) [inline contract/data/contract_cron.xml]:
```suggestion
<?xml version="1.0" encoding="UTF-8" ?>
```

File: contract/data/contract_renew_cron.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding='UTF-8' ?>
```
Comment (newtratip) [inline contract/data/contract_renew_cron.xml]:
```suggestion
<?xml version="1.0" encoding="UTF-8" ?>
```

File: contract/data/mail_template.xml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" ?>
```
Comment (newtratip) [inline contract/data/mail_template.xml]:
```suggestion
<?xml version="1.0" encoding="utf-8" ?>
```

File: contract/README.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
+================================
```
Comment (newtratip) [inline contract/README.rst:1]: Please generate readme follow this https://github.com/OCA/maintainer-tools/blob/master/README.md#readme-generator

File: contract/report/report_contract.xml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                                <strong>Responsible: </strong>
+                                <p t-field="o.user_id" />
+                                <strong>Contract: </strong>
+                                <p t-field="o.code" />
+                            </div>
+                        </div>
+                        <div class="row" id="invoice_info">
+                            <t t-set="total" t-value="0" />
+                            <div class="col-12">
+                                <t t-set="total" t-value="0" />
```
Comment (newtratip) [inline contract/report/report_contract.xml]: `<t t-set="total" t-value="0" />` is duplicated with above line 31, Can you remove one line ?

File: contract/security/contract_security.xml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?xml version='1.0' encoding='utf-8' ?>
```
Comment (newtratip) [inline contract/security/contract_security.xml]:
```suggestion
<?xml version="1.0" encoding="utf-8" ?>
```

File: contract/wizards/contract_manually_create_invoice.xml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                            attrs="{'invisible': [('contract_to_invoice_count', '=', 0)]}"
+                        >
+                            <field name="contract_to_invoice_count" />
+                            <span
+                                attrs="{'invisible': [('contract_to_invoice_count', '&gt;', 1)]}"
+                            >
+                                contract to invoice
+                            </span>
+                            <span
+                                attrs="{'invisible': [('contract_to_invoice_count', '&lt;', 1)]}"
```
Comment (newtratip) [inline contract/wizards/contract_manually_create_invoice.xml]:
```suggestion
                                attrs="{'invisible': [('contract_to_invoice_count', '&lt;=', 1)]}"
```

**Review by newtratip — COMMENTED** [review]
Some review

**Comment by newtratip** [comment]
**Functional Test**
Terminate Contract
1. Click Terminate Contract button
2. Fill Termination Date, Termination Reason, Termination Comment
3. Click Terminate Contract and click `OK` on Confirmation dialog box
4. Confirmation dialog box still show after contract terminated

My video
Peek 2021-02-15 15-48

**Comment by dsolanki-initos** [comment]
Is it possible to get merge this PR as it is depended on other modules.

**Comment by pedrobaeza** [comment]
@ozono all changes are merged in previous versions, so this only gets to be up to date with them and attend comments.

**Comment by ozono** [comment]
Ok. I'll do it ASAP.

**Comment by ozono** [comment]
All comments and suggestions have been resolved / taken into account. Thanks for the work done for V 13.0 module.

File: contract/readme/USAGE.rst [hunk 8]
```diff
@@ -23,9 +23,4 @@
 #. Contract templates can be created from the Configuration -> Contracts -> Contract Templates menu.
    They allow to define default journal, price list and lines when creating a contract.
    To use it, just select the template on the contract and fields will be filled automatically.
-
-* Contracts appear in portal to following users in every contract:
-
-.. image:: ../static/src/screenshots/portal-my.png
```
Comment (pedrobaeza) [inline contract/readme/USAGE.rst]: Why removing these images?

Comment (ozono) [reply contract/readme/USAGE.rst]: I think that images was meaningful when new feature was introduced. Moreover: images are not present un V. 13.0 modules (there are broken links). But, maybe, I'm wrong... If you don't agree I can take new snapshots and fix this.

Comment (pedrobaeza) [reply contract/readme/USAGE.rst]: The idea is to showcase features in README, so they are relevant, but it's a "bug" that they weren't present in 13.0. @victoralmau can you please recover them?

Comment (ozono) [reply contract/readme/USAGE.rst]: Ok. I'll recover them too.

Comment (ozono) [reply contract/readme/USAGE.rst]: Done @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]

File: contract/readme/USAGE.rst (hunk 8 above)
Comment (victoralmau) [reply contract/readme/USAGE.rst]: I have created these PR https://github.com/OCA/contract/pull/659 to add portal images in 13.0

**Pull request merged**